master/internal/command/command.go

```diff
@@ -40,6 +42,25 @@ const terminatedDuration = 24 * time.Hour
 // should stop and garbage collect its state.
 type terminateForGC struct{}
 
+func enrichState(state taskv1.State) taskv1.State {
```

Comment:
the disconnect between what pending means internally and what is means externally (pending isnt a pending state) is probably going to get really confusing really fast. any thoughts?

Reply:
this is the only thing i'm really concerned about, maybe we can chat offline about a timeline for cleaning up the states that are now dead as far as the API is concerned.